### PR TITLE
nproc: replace num_cpus crate with thread::available_parallelism

### LINKED ISSUE #4351: Replace `num_cpus` crate with `std::thread::available_parallelism`
`nproc` uses the `num_cpus` crate, but this functionality has been added to `std` in the form of [`std::thread::available_parallelism`](https://doc.rust-lang.org/stable/std/thread/fn.available_parallelism.html). I suggest we move over. I _think_ there's no problem in doing this, though there are slight differences between `num_cpus` and `std`: https://github.com/seanmonstar/num_cpus/issues/119.

### PR DESCRIPTION
**Pull request by rgasnix**
Closes #4351

### CODE REVIEW

**Comment by sylvestre** [comment]
Looks good
Should we check the error ? 
(in which cases can we get any?)

**Comment by rgasnix** [comment]
From what i have read here [std::thread::available_parallelism](https://doc.rust-lang.org/stable/std/thread/fn.available_parallelism.html) a error will, but it's not limited to, return in the following cases:
- If the amount of parallelism is not known for the target platform.
- If the program lacks permission to query the amount of parallelism made available to it.  
I'm not sure if at any point we will reach those scenarios, but I'm doing a research to see others pitfall we may find.

**Comment by tertsdiepraam** [comment]
I think GNU nproc just returns 1 processor if all methods fail: https://github.com/coreutils/gnulib/blob/f2b8ad906cbc861552a64e3804e5851c323cbac7/lib/nproc.c#L339

Not sure I agree with that, but that's probably what we should do for compatibility.

**Comment by rgasnix** [comment]
So for compatibility reasons it's better to return 1 in case it falls in an error?

If we agree with that I will do the changes.

**Comment by tertsdiepraam** [comment]
Yes I think so, but let's also include a comment on the GNU behaviour for context.

File: src/uu/nproc/src/nproc.rs [hunk 1]
```diff
@@ -73,16 +73,27 @@ pub fn uumain(args: impl uucore::Args) -> UResult<()> {
                 // If OMP_NUM_THREADS=0, rejects the value
                 let thread: Vec<&str> = threadstr.split_terminator(',').collect();
                 match &thread[..] {
-                    [] => thread::available_parallelism()?.get(),
+                    // In some cases, thread::available_parallelism() may return an Err
+                    // In this case, we will return 1 (like GNU)
```
[the remark this answers is not in the document]
Comment (tertsdiepraam) [reply src/uu/nproc/src/nproc.rs:77]: You could also wrap the call to `available_parallelism` into a separate function and document it there:
```rust
/// Document the behaviour here.
fn available_parallelism() -> usize {
    match std::thread::available_parallelism() {
        Ok(n) => n.get(),
        Err(_) => 1
    }
}
```

**Review by tmccombs — APPROVED** [review]

File: src/uu/nproc/src/nproc.rs [hunk 2]
```diff
@@ -127,21 +127,30 @@ fn num_cpus_all() -> usize {
...
+        available_parallelism()
     } else if nprocs > 0 {
         nprocs as usize
     } else {
         1
     }
 }
 
-// Other platforms (e.g., windows), num_cpus::get() directly.
+// Other platforms (e.g., windows), thread::available_parallelism()?.get() directly.
```
Comment (tertsdiepraam) [inline src/uu/nproc/src/nproc.rs:138]: This still refers to `std::thread` instead of the new function

**Review by tertsdiepraam — COMMENTED** [review]
Looks good! Just one more nit

**Review by tertsdiepraam — APPROVED** [review]
All good now! Thanks!

**Comment by rgasnix** [comment]
Of course, I'll pay more attention in the next PRs and sorry for the inconvenience.

**Comment by sylvestre** [comment]
not a big deal :) it is just best practice :)

**Pull request merged**